test/basic.jl

```diff
@@ -221,21 +221,21 @@ end
 
     # Ensure passing compat bounds works
     dependencies = [
-        Dependency(PackageSpec(name="libLLVM_jll", version=v"9")),
+        Dependency(PackageSpec(name="libLLVM_jll"), compat="=9.0.0"),
     ]
     dict = build_project_dict("Clang", v"9.0.1+2", dependencies)
     @test dict["compat"]["julia"] == "1.0"
     @test dict["compat"]["libLLVM_jll"] == "=9.0.0"
 
     dependencies = [
-        Dependency(PackageSpec(name="libLLVM_jll", version="8.3-10")),
+        Dependency(PackageSpec(name="libLLVM_jll"), compat="8.3 - 10"),
     ]
     dict = build_project_dict("Clang", v"9.0.1+2", dependencies)
     @test dict["compat"]["julia"] == "1.0"
-    @test dict["compat"]["libLLVM_jll"] == "8.3-10"
+    @test dict["compat"]["libLLVM_jll"] == "8.3 - 10"
```

Comment:
Note that the old compat string was invalid, and hence so was this test.

Also note that the *new* compat string only is supported in Julia >= 1.4 (IIRC), so I think my PRs should either outright reject it, or at least reject it if the `julia_compat` strings allows for older Julia versions. Thoughts?